### PR TITLE
Replace `imageMutex` with `Locker` pkg

### PR DESCRIPTION
**Pull request by cpuguy83**

### CODE REVIEW

**Comment by vdemeester** [comment]
LGTM 🐸

File: graph/mutex.go [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-import "sync"
-
-// imageMutex provides a lock per image id to protect shared resources in the
-// graph. This is only used with registration but should be used when
-// manipulating the layer store.
-type imageMutex struct {
-	mus map[string]*sync.Mutex // mutexes by image id.
-	mu  sync.Mutex             // protects lock map
-
-	// NOTE(stevvooe): The map above will grow to the size of all images ever
```
Comment (runcom) [inline graph/mutex.go]: pkg/locker should even handle this :)

**Comment by runcom** [comment]
LGTM

**Comment by runcom** [comment]
Windows fail was unrelated :)

**Pull request merged**